### ISSUE TITLE
clang already has mature C++20 support

### ISSUE THREAD

**Issue by 2dpdlja496or43iq** [Sygmei/ObEngine#445]
`experimental` is no longer needed and will no longer work

```
#include <experimental/source_location>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

`experimental::source_location` will no longer work, it's `std::source_location` now.

**Comment by Sygmei** [comment]
I tried it out as you can see here and it failed : https://github.com/ObEngine/ObEngine/actions/runs/4282915582/jobs/7457788613

If you happen to know how I can fix it, I'd be interested !

**Comment by 2dpdlja496or43iq** [comment]
> I tried it out as you can see here and it failed : https://github.com/ObEngine/ObEngine/actions/runs/4282915582/jobs/7457788613
> 
> If you happen to know how I can fix it, I'd be interested !

This is APPLE CLANG, not the normal CLANG

**Comment by Sygmei** [comment]
Oh right ! I'll put that section behind a ifdef

**Comment by Sygmei** [comment]
Fixed for Clang 16 : [5cf4aeb10ce7fd410d1c7677316c6c7afef69555](https://github.com/ObEngine/ObEngine/commit/5cf4aeb10ce7fd410d1c7677316c6c7afef69555)
